### PR TITLE
feat(protocol-designer): update heater shaker form fields

### PR DESCRIPTION
**Pull request by ncdiehl11**
# Overview

This PR overhauls the new PD heater shaker form. I consolidate shake duration form field (minutes and seconds) into a single time field. I also fix the function of the HS form toggle input fields and update errors, maskers, and default values for the form. I also add form-level HS form errors to catch required fields when they are toggled but no input is given.

Closes AUTH-864, Closes AUTH-966

## Test Plan and Hands on Testing

- create or import a PD protocol with heater shaker step
- toggle each form field and test various inputs, checking that errors are properly caught

## Changelog

- implement consolidated `heaterShakerTimer` form field
- fix behavior or heater shaker toggle input form fields
- add errors for heater shaker form
- abstract `getTimeFromForm` utility to work with heater shaker form in addition to pause form
- update heater shaker form field copy

## Review requests

- see test plan

## Risk assessment

medium. Modifies behavior of heater shaker form to args. I checked tests and exported protocols with the new consolidated `timer` field, and all looks correct.

### CODE REVIEW

**Comment by jerader** [comment]
ran into a small bug 

https://github.com/user-attachments/assets/3defc88b-5c49-4325-aaeb-8290d4d3ba34

File: protocol-designer/src/atoms/ToggleButton/index.tsx [hunk 1]
```diff
@@ -59,7 +59,9 @@ export function ToggleButton(props: ToggleButtonProps): JSX.Element {
       css={props.toggledOn ? TOGGLE_ENABLED_STYLES : TOGGLE_DISABLED_STYLES}
       {...buttonProps}
     >
-      <Icon name={iconName} height="1rem" />
+      <Flex>
+        <Icon name={iconName} height="1rem" />
```
Comment (koji) [inline protocol-designer/src/atoms/ToggleButton/index.tsx]:
```suggestion
        <Icon name={iconName} size="1rem" />
```

File: protocol-designer/src/steplist/formLevel/stepFormToArgs/heaterShakerFormToArgs.ts [hunk 2]
```diff
@@ -36,13 +45,7 @@ export const heaterShakerFormToArgs = (
...
-    timerMinutes:
-      formData.heaterShakerTimerMinutes != null
-        ? parseInt(formData.heaterShakerTimerMinutes)
-        : null,
-    timerSeconds:
-      formData.heaterShakerTimerSeconds != null
-        ? parseInt(formData.heaterShakerTimerSeconds)
-        : null,
+    timerMinutes: isNullTime ? null : minutes,
+    timerSeconds: isNullTime ? null : seconds,
```
Comment (koji) [inline protocol-designer/src/steplist/formLevel/stepFormToArgs/heaterShakerFormToArgs.ts:49]: if seconds === 0 then, we don't need to use minutes' value?

Comment (ncdiehl11) [reply protocol-designer/src/steplist/formLevel/stepFormToArgs/heaterShakerFormToArgs.ts:49]: we pass null for both values only if both values are null. If one is non-null, we pass 0 or the actual value (consistent with tests)

**Review by jerader — APPROVED** [review]
lgtm! nice work! i also tested and the timer command was generated as expected

Only thing is the toggles are smaller for some reason this time around. But the first time i looked at your branch, the toggles were the correct size, so idk what happened 🥲 

Screenshot 2024-10-23 at 16 58 11

**Comment by koji** [comment]
Screenshot 2024-10-23 at 4 58 37 PM

**Review by koji — APPROVED** [review]

**Pull request merged**